File: parsers/ruby_plain_json/test_json.rb
#!/usr/bin/ruby

require 'json'

f = ARGV[0]

o = nil

begin
    puts(f)
    json = File.read(f)
    o = JSON.parse(json, {:quirks_mode => true})
    p o
    puts(o)
    if o == nil
        puts("incomplete")
        exit 1
    else
        puts("valid")
        exit 0
    end
rescue JSON::ParserError => e
    puts(e)
    puts("invalid")
    exit 2
end

File: parsers/javascript_json/test_json.js
var fs = require('fs');
var path = process.argv[2];

try {
	var data = fs.readFileSync(path);
	var json = JSON.parse(data);
	//console.log(json)
	console.log("valid");
	process.exit(0);
} catch (e) {
	console.log("--", e.message);
	if (e.message.includes("Unexpected token") || e.message.includes("Unexpected number") || e.message.includes("Unexpected string")) {
		console.log("invalid");
		process.exit(2);
	}
	console.log("incomplete");
	process.exit(1);
}
process.exit(0);
